tests/ConnectionTest.php

```diff
@@ -84,17 +83,19 @@ function (\Exception $error) {
         );
     }
 
-    /**
-     * @expectedException React\MySQL\Exception
-     * @expectedExceptionMessage Can't send command
-     */
-    public function testPingWithoutConnectThrows()
+    public function testPingWithoutConnectRejects()
     {
         $options = $this->getConnectionOptions();
         $loop = \React\EventLoop\Factory::create();
         $conn = new Connection($loop, $options);
 
-        $conn->ping(function () { });
+        $conn->ping()->then(
+            $this->expectCallableNever(),
+            function (\Exception $error) {
+                $this->assertInstanceOf('React\MySQL\Exception', $error);
+                $this->assertSame('Can\'t send command', $error->getMessage());
+            }
+        );
```

Comment:
Assertions should not be done inside rejection callbacks because the exceptions thrown there get swallowed.

For example, the following test will succeed:

```php
(new Promise(function ($res, $rej) {
    $rej('foo');
}))->then(
    $this->expectCallableNever(),
    function (\Exception $error) {
        $this->assertInstanceOf('React\MySQL\Exception', $error);
        $this->assertSame('Can\'t send command', $error->getMessage());
    }
);
```

Reply:
I agree and also see potential issues here. I've decided to keep the test structure as-is, as this also affects the changes in #61. Do you think it makes sense to keep this as-is for now and address this in a follow-up PR once #4 is completed? :shipit:

Reply:
If `ping()` is synchronous, the assertions could be moved outside the callback.

Another option would be to ensure the promise does not get rejected by chaining an addtionional `->then(null, $this->expectCallableNever())`:

```php
->then(
    $this->expectCallableNever(),
    function (\Exception $error) {
        $this->assertInstanceOf('React\MySQL\Exception', $error);
        $this->assertSame('Can\'t send command', $error->getMessage());
    }
)
->then(null, $this->expectCallableNever());

Reply:
Sounds reasonable, but I'd argue that semantically this may seem confusing. As an alternative, I've just updated this PR to use the `done()` method and have verified that this actually yields an error if the assertion does not match :shipit: 

That being said, I consider this a temporary solution and we should look into this again in a follow-up PR to also address this issue for the `query()` method and others :+1: